arcaflow_plugin_stressng/stressng_schema.py

```diff
@@ -157,6 +566,12 @@ class StressNGParams:
                     schema.name("VM Stressor Parameters"),
                     schema.description("Parameters for running the vm stressor"),
                 ],
+                typing.Annotated[
+                    MmapStressorParams,
+                    annotations.discriminator_value("mmap"),
```

Comment:
Shouldn't the value in line 571 be `Stressors.MMAP`, and similar for the other discriminator values, here?

Reply:
Maybe? I'm not entirely sure how this part works. I'll test it.

Reply:
That change breaks the build.

Reply:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/arcaflow_plugin_sdk/schema.py", line 6837, in _resolve_list_annotation
    cls._resolve_abstract_type(
  File "/usr/local/lib/python3.9/site-packages/arcaflow_plugin_sdk/schema.py", line 6324, in _resolve_abstract_type
    result = cls._resolve(t, type_hints, path, scope)
  File "/usr/local/lib/python3.9/site-packages/arcaflow_plugin_sdk/schema.py", line 6389, in _resolve
    return cls._resolve_annotated(t, type_hints, path, scope)
  File "/usr/local/lib/python3.9/site-packages/arcaflow_plugin_sdk/schema.py", line 6771, in _resolve_annotated
    underlying_t = cls._resolve(args[0], args_hints[0], path, scope)
  File "/usr/local/lib/python3.9/site-packages/arcaflow_plugin_sdk/schema.py", line 6387, in _resolve
    return cls._resolve_union(t, type_hints, path, scope)
  File "/usr/local/lib/python3.9/site-packages/arcaflow_plugin_sdk/schema.py", line 7003, in _resolve_union
    raise BadArgumentException(
arcaflow_plugin_sdk.schema.BadArgumentException: Invalid discriminator value type: <class 'str'>, the value type has been previously set to <enum 'Stressors'>. Please make sure all your discriminator values have the same type.
```

Reply:
Yeah, ARCA input based on schema enums is weird ... it delivers the ENUM in the plugin step parameters, but in the YAML input the ENUM really only restricts the allowed range of int or str values ... you can't actually specify the ENUM name as a token. I always thought this was really weird and counterintuitive ...

Reply:
So, `annotations.discriminator_value(Stressors.MMAP.name)` or `annotations.discriminator_value(str(Stressors.MMAP))`, then?  😇

Reply:
`.name` won't work unless it's identical to the value, and I think the default stringification uses `.name`, too. You need the `.value`. E.g.,

```
class Foo(Enum):
    XYZZY = "plugh"
```

`str(Foo.XYZZY) == 'Foo.XYZZY` and `Foo.XYZZY.name` == `XYZZY`, which won't parse; you need `Foo.XYZZY.value` (`plugh`).

Reply:
> you need `Foo.XYZZY.value`

Ah!  Yes!!  Right.  Thanks.  🙂